### ISSUE TITLE
"space-after-keywords: never" + "space-before-blocks: always" incompatible with try & `else

### ISSUE THREAD

**Issue by Cortys** [eslint/eslint#4006]
Using eslint 1.5.1. This issue might be related to #1338.

I want to configure the following code style:
```js
function() {}

if(true) {}
else {}

try {}
catch(e) {}
```

Obviously I have to enforce spaces before blocks via `space-before-blocks: always`. To additionally forbid things like `if () {}` I only found the `space-after-keywords: never` rule.

This however makes it impossible to write `try` and `else` statements:
```js
try {} // space-after-keywords: Keyword "try" must not be followed by whitespace.
   ^

try{} // space-before-blocks: Missing space before opening brace.
   ^

```

I see that this configuration is kind of contradictory. Is there another way to achieve what I want?

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
We probably need to ensure `space-after-keywords` does not apply when there's a block immediately following it. 

@mysticatea I think you just fixed something similar to this?

**Comment by mysticatea** [comment]
I think this issue is almost duplicate of #1338.
I mentioned similar thing in #3821, but not fixed.

**Comment by aparajita** [comment]
There's another bug in the implementation of `space-after-keywords` when using allman braces:

```js
do
{
    doSomething();
}
```

It complains that `Keyword "do" must not be followed by a newline.` Huh? If a newline isn't considered space after a keyword, I don't know what is.

**Comment by nzakas** [comment]
@aparajita can you open a separate issue for that?

**Comment by aparajita** [comment]
Eventually, when I get some time.
